code/renderers/vue3/template/stories/ReactiveArgs.stories.js

```diff
@@ -0,0 +1,47 @@
+import { expect } from '@storybook/jest';
+import { global as globalThis } from '@storybook/global';
+import { within, userEvent } from '@storybook/testing-library';
+import { UPDATE_STORY_ARGS, STORY_ARGS_UPDATED, RESET_STORY_ARGS } from '@storybook/core-events';
+import ReactiveArgs from './ReactiveArgs.vue';
+
+export default {
+  component: ReactiveArgs,
+  argTypes: {
+    // To show that other props are passed through
+    backgroundColor: { control: 'color' },
+  },
+};
+
+export const ReactiveTest = {
+  args: {
+    label: 'Button',
+  },
+  // test that args are updated correctly in rective mode
+  play: async ({ canvasElement, id }) => {
+    const channel = globalThis.__STORYBOOK_ADDONS_CHANNEL__;
+    const canvas = within(canvasElement);
+
+    const updatedPromise = new Promise((resolve) => {
+      channel.once(STORY_ARGS_UPDATED, resolve);
+    });
+
+    const reactiveButton = await canvas.getByRole('button');
+    await expect(reactiveButton).toHaveTextContent('Button 0');
+
+    await userEvent.click(reactiveButton); // click to update the label to increment the count + 1
+    await channel.emit(UPDATE_STORY_ARGS, {
+      storyId: id,
+      updatedArgs: { label: 'updated' },
+    });
+    await updatedPromise;
+    await expect(canvas.getByRole('button')).toHaveTextContent('updated 1');
+
+    await userEvent.click(reactiveButton); // click to update the label to increment the count + 1
+    await expect(reactiveButton).toHaveTextContent('updated 2');
+
+    await channel.emit(RESET_STORY_ARGS, { storyId: id });
+    await new Promise((resolve) => {
+      channel.once(STORY_ARGS_UPDATED, resolve);
+    });
```

Comment:
I think you should move this to the beginning of the play function.
I was confused that the result was "Button 2", because I thought it meant the args had never updated, but then I realised it was because they were reset. if you reset at the beginning that confusion won't be a problem

Reply:
yes you are better in the beginning.